resources/command_line_tools.rb

```diff
@@ -2,45 +2,48 @@
 
 provides :command_line_tools
 
+property :name, String, default: ''
 property :compile_time, [true, false],
   description: 'Install the Xcode Command Line Tools at compile time.',
   default: false, desired_state: false
 
 action :install do
-  command_line_tools = CommandLineTools.new
-
-  file 'create sentinel file' do
+  file 'create demand file' do
     path '/tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress'
     group 'wheel'
+    not_if { ::File.exist?('/Library/Developer/CommandLineTools/usr/lib/libxcrun.dylib') }
+    notifies :run, 'execute[install command line tools]', :immediately
```

Comment:
So if we tell the `execute[install command line tools]` block to run immediately, would that skip over line 18 where we create a new `CommandLineTools` object?

Reply:
it won't ship - the execution path when both "Chef DSL" and Ruby co-exist is confusing. TL;DR is that `immediately` means not `delayed` until the end of the Chef run: https://docs.chef.io/resource_common/#notifications